### DISCUSSION TITLE
GSSOC'23 How to contribute to Web Resources Hub? 🚀🚀

### DISCUSSION THREAD

**Discussion by jayk-gupta** [jayk-gupta/web-resources-project#59] [Announcements]
## Welcome to **Web Resources Hub** ✨ , an exciting open-source project that has been selected for **GirlScript Summer of Code [GSSoC'23]**!

![Screenshot 2023-05-21 114628](https://github.com/jayk-gupta/web-resources-project/assets/100681165/a249a55e-ffd4-4ccc-a37b-1f89a0d4be0a)

We are thrilled to have you on board and are looking forward to your valuable contributions. This guide will provide you with **comprehensive guidelines** on how to contribute effectively to this project during this amazing event.

# Getting Started
Before going to contribute please check this guide:

## Understand the Project:

Familiarize yourself with the project's objectives, architecture, and existing codebase. This will enable you to make informed contributions that align with the project's goals.
Choose an Issue/Create an issue:

## Issue
Create an issue or work on the existing issue if you're assigned. Also, if you're in GSSOC'23 please add "I'm in GSSOC'23" so that we can add the GSSOC'23 label on that particular issue/PR.
Work on Your assigned issue first:

## Prioritizing Issues
We kindly request that you prioritize working on your assigned issue at this time. It is highly recommended that you begin by creating an issue before proceeding to create a pull request.
By doing so, you can outline the problem or task at hand and facilitate a smoother workflow. Once the issue is established, please proceed to implement the necessary changes, address any bugs, or incorporate new features to effectively resolve the selected issue/task.

## Commit Your Changes:

Once you have made significant progress or completed your contribution, commit your changes with a descriptive commit message. It is advisable to make frequent, small commits to make the review process more manageable. Please check [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0)
Push Your Changes:

## Pushing the changes:

Push your local branch to your GitHub repository using the git push command. This will make your changes accessible for review and integration into the main project.

## Create a Pull Request:

Navigate to the original Web Resources Hub repository on GitHub and create a pull request from your forked repository. Provide a clear and concise description of your changes, referencing the issue/task you addressed. Mention any relevant tests or documentation that you have added or updated.
Review and Collaboration
Engage in Discussions:

Engage in active discussions on the pull request, addressing any questions, suggestions, or feedback from the project maintainers or the community. Collaborate and iterate on your contribution based on the feedback received.

## Make Necessary Changes:

Make the requested changes and push them to your branch. The pull request will update automatically, reflecting the new changes.

## Resolve Conflicts:

In case there are conflicts with the main project's branch, resolve them by merging or rebasing your branch onto the latest changes from the main project.

## Tests and Documentation:

Ensure that your contribution includes appropriate tests to validate the functionality and any relevant updates to the project's documentation.

## Get Your Contribution Merged:

Once your contribution has been reviewed, approved, and meets the project's requirements, a project maintainer will merge your changes into the main codebase.
Congratulations on successfully contributing to Web Resources Hub! 🎉🎉

We already have [Contributing guide[(https://github.com/priyankarpal/ProjectsHut/blob/main/contributing.md](https://github.com/jayk-gupta/web-resources-project/blob/main/CONTRIBUTING.md)) you can check it. 
Also, don't forget to check the [README](https://github.com/jayk-gupta/web-resources-project/edit/main/README.md)

All the best! 🚀❤️

**Comment by Esha-in-arena** [comment]
hey i have created an issue without adding I'm in GSSOC'23  now I am unable to add label and edit the issue what shall I do

**Comment by Jranii** [comment]
I'm in GSSOC'23. Kindly add the label GSSOC'23 to my issue #311.
